src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -743,22 +745,34 @@ def get_keyring(self, rgw_id: str) -> str:
         return keyring
 
     def create_realm_zonegroup_zone(self, spec: RGWSpec, rgw_id: str) -> None:
-        if utils.get_cluster_health(self.mgr) != 'HEALTH_OK':
-            raise OrchestratorError('Health not ok, will try again when health ok')
-
         # get keyring needed to run rados commands and strip out just the keyring
         keyring = self.get_keyring(rgw_id).split('key = ', 1)[1].rstrip()
 
-        # We can call radosgw-admin within the container, cause cephadm gives the MGR the required keyring permissions
+        # We can call radosgw-admin within the container, cause
+        # cephadm gives the MGR the required keyring permissions.
 
-        def get_realms() -> List[str]:
-            cmd = ['radosgw-admin',
-                   '--key=%s' % keyring,
-                   '--user', 'rgw.%s' % rgw_id,
-                   'realm', 'list',
-                   '--format=json']
+        # Include a timeout so that we don't block indefinitely if there is a pool
+        # unavailable.
+
+        def radosgw_admin(args: List[str]) -> Tuple[bytes, bytes, int]:
+            cmd = [
+                'timeout', RADOSGW_ADMIN_TIMEOUT,
+                'radosgw-admin',
+            ] + args
             result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
```suggestion
            try:
               result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, timeout=RADOSGW_ADMIN_TIMEOUT)
            except subprocess.TimeoutExpired as e:
                raise OrchestratorError(f'Timed out calling radosgw-admin; will try again later: {e}')
```

use subprocess timeout?